packages/vm/src/evm/opcodes/codes.ts

```diff
@@ -322,5 +325,36 @@ export function getOpcodesForHF(common: Common): OpcodeList {
     opcodeBuilder[key].fee = common.param('gasPrices', opcodeBuilder[key].name.toLowerCase())
   }
 
+  if (customOpcodes) {
+    for (const _code of customOpcodes) {
+      const code = <any>_code
+      if (code.logicFunction === undefined) {
+        delete opcodeBuilder[code.opcode]
+        continue
+      }
+
+      // Sanity checks
+      if (code.opcodeName === undefined || code.baseFee === undefined) {
+        throw new Error(
+          'Custom opcode does not have the required values: opcodeName and baseFee are required'
+        )
+      }
+      const entry = {
+        [code.opcode]: {
+          name: code.opcodeName,
+          isAsync: true,
+          dynamicGas: code.gasFunction !== undefined,
+          fee: code.baseFee,
+        },
+      }
+      opcodeBuilder = { ...opcodeBuilder, ...entry }
+      if (code.gasFunction) {
+        dynamicGasHandlers.set(code.opcode, code.gasFunction)
```

Comment:
I love this so much to see how these things play so nicely together close to everytime we do a profound refactoring which was initially meant to solve something completely different! 🥳

Thanks again for this great work! ❤️